front/components/assistant_builder/actions/ProcessAction.tsx

```diff
@@ -410,16 +409,23 @@ export function ActionProcess({
         </Hoverable>
         .
       </div>
-
       <DataSourceSelectionSection
         owner={owner}
         dataSourceConfigurations={actionConfiguration.dataSourceConfigurations}
         openDataSourceModal={() => {
           setShowDataSourcesModal(true);
         }}
+        onSave={(dsConfigs) => {
+          setEdited(true);
+          updateAction((previousAction) => ({
+            ...previousAction,
+            dataSourceConfigurations: dsConfigs,
+          }));
+        }}
         viewType="documents"
       />
 
+      {/* TODO(TAF): Remove this once tag filtering is rolled out */}
```

Comment:
I _think_ we want to wrap both UIs in the feature flag to make sure we can't have double UI: 
- if feature flag -> new UI and not old UI
- if not feature flag --> old and not new
WDYT?